### ISSUE TITLE
Solidity compiler fails to work on some Mac setups

### ISSUE THREAD

**Issue by Nashatyrev** [ethereum/ethereumj#367]
Most likely the problem is missing some dylibs which are installed on some setups but missed on others. 
One known is something like json-rpc.dylib

**Comment by Nashatyrev** [comment]
The best way to test is to use clean OS installation

**Comment by Nashatyrev** [comment]
@loxal didn't you see any output from `CompilerTest` ?

**Comment by loxal** [comment]
Exactly these are the both tests that failing on my machine related to:
```
        SolidityCompiler.Result res = SolidityCompiler.compile(
                contract.getBytes(), true, SolidityCompiler.Options.ABI, SolidityCompiler.Options.BIN, SolidityCompiler.Options.INTERFACE);
```
I’ve debugged a bit and it turns out that 

```
        ParallelReader error = new ParallelReader(process.getErrorStream());
        ParallelReader output = new ParallelReader(process.getInputStream());
``` 
both eventually return an empty string. I’ve provided you with corresponding JUnit tests already to catch this case with a **failing assertion** instead of **undefined behavior** (failing test because of some exception).

**Comment by Nashatyrev** [comment]
@loxal Can you please try copying those files to a dir: https://github.com/ethereum/ethereumj/tree/develop/ethereumj-core/src/main/resources/native/mac/solc
and running `solc`. What will it say?
I guess you'll need to `chmod +x solc`

**Comment by loxal** [comment]
The solc version in the repo is v1.6.x which is quite old. The most recent one is “Version: 0.3.1-33b95459/Release-Darwin/appleclang/Interpreter”. With this version the ≈10 tests that were broken all pass :+1:  

Those were the broken tests that I was talking about. Actually on other branches they were only two related tests. On `stage` there were some more tests broken. Now all 717 tests pass.

**Comment by loxal** [comment]
The solc version I’ve provided is the latest one you can get from brew, Mac’s package manager.

**Comment by romanman** [comment]
@loxal : we are going to release the 
ethereumj core with solc `0.2.2`, 
can you please test our fix for mac
libs support

**Comment by Nashatyrev** [comment]
This should be fixed in 1.2.0
